overlord/snapstate/check_snap.go

```diff
@@ -175,8 +176,16 @@ var featureSet = map[string]bool{
 // https://systemd.io/UIDS-GIDS.html
 // https://docs.docker.com/engine/security/userns-remap/
 // https://github.com/lxc/lxd/blob/master/doc/userns-idmap.md
-var supportedSystemUsernames = map[string]uint32{
-	"snap_daemon": 584788,
+type systemUsername struct {
+	Id uint32
+	// List of snap IDs which are allowed to declare this user or nil if no
+	// such restriction exists
+	AllowedSnapIds []string
+}
+
+var supportedSystemUsernames = map[string]systemUsername{
```

Comment:
my suggestion is to move this and the type above for now into  snapstate/policy/system_usernames.go as publicly exported, and import it for use here from there

Reply:
my suggestion breaks because of circular dependencies which I didn't consider, the new suggestion discussed is simply to introduce:  snap/system_usernames.go